components/style/properties/longhand/effects.mako.rs

```diff
@@ -1112,6 +1151,40 @@ ${helpers.predefined_type("opacity",
 
             computed_value::T(Some(result))
         }
+        #[inline]
+        fn to_specified_value(computed: &computed_value::T, context: &UncomputeContext) -> Self {
+            SpecifiedValue(computed.0.as_ref().map(|computed| {
+                let mut result = vec!();
+                for operation in computed {
+                    match *operation {
+                        computed_value::ComputedOperation::Matrix(ref matrix) => {
+                            result.push(SpecifiedOperation::Matrix(*matrix));
+                        }
+                        computed_value::ComputedOperation::Translate(ref tx, ref ty, ref tz) => {
+                            result.push(SpecifiedOperation::Translate(TranslateKind::Translate,
```

Comment:
Is `TranslateKind::Translate` the right thing to use here?  The specified value would round trip back to a computed value properly (since we always pass down the third component), but it wouldn't serialize properly.  I think this should either be `translate3d` always, or maybe based on whether `tz` is 0.

Incidentally, I don't think it's correct that the specified value loses whether it was a scale(), or scaleZ(), etc. when parsed.  That information should be preserved.  (It should also be preserved in the computed value, according to the spec.)